### ISSUE TITLE
Crash on <date day="??"

### ISSUE THREAD

**Issue by cabo** [ietf-tools/xml2rfc#794]
```
            <date year="2017" month="September" day="??"/>
```

in a reference is enough to crash xml2rfc:

```
Traceback (most recent call last):
  File "/Users/cabo/bin/xml2rfc", line 8, in <module>
    sys.exit(main())
  File "/Users/cabo/lib/python3.9/site-packages/xml2rfc/run.py", line 737, in main
    xmlrfc.tree = prep.prep()
  File "/Users/cabo/lib/python3.9/site-packages/xml2rfc/writers/preptool.py", line 216, in prep
    tree = self.dispatch(self.selectors)
  File "/Users/cabo/lib/python3.9/site-packages/xml2rfc/writers/base.py", line 1925, in dispatch
    func(e, e.getparent())
  File "/Users/cabo/lib/python3.9/site-packages/xml2rfc/writers/preptool.py", line 419, in check_attribute_values
    if i and not i.isdigit() and not i==self.get_attribute_defaults(c.tag)[a]:
KeyError: 'day'
```

**Comment by cabo** [comment]
Clearly the code

```
        integer_attributes = {
            'rfc':      ('number', 'version', ),
            'date':     ('year', 'day', ),
            'dl':       ('indent', ),
            'format':   ('octets', ),
            'ol':       ('indent', ),
            't':        ('indent', ),
            'ul':       ('indent', ),
        }
        tags = integer_attributes.keys()
        for c in e.iter(tags):
            for a in integer_attributes[c.tag]:
                i = c.get(a)
                if i and not i.isdigit() and not i==self.get_attribute_defaults(c.tag)[a]:
                    self.err(c, 'Expected <%s> attribute "%s" to be a non-negative integer, but found "%s"' % (c.tag, a, i))

```

is assuming there to be a default value for "day", which seems to be not available.
Maybe the ` i==self.get_attribute_defaults(c.tag)[a]` can be coded more robustly.

**Comment by kesara** [comment]
Fixed with #795
